Framework/DataHandling/src/LoadILLSANS.cpp

```diff
@@ -514,11 +572,16 @@ size_t LoadILLSANS::loadDataFromTubes(NeXus::NXInt &data, const std::vector<doub
  * Create a workspace without any data in it
  * @param numberOfHistograms : number of spectra
  * @param numberOfChannels : number of TOF channels
+ * @param type : type of the multichannel workspace (TOF (default) or Kinetic)
  */
-void LoadILLSANS::createEmptyWorkspace(const size_t numberOfHistograms, const size_t numberOfChannels) {
-  m_localWorkspace =
-      WorkspaceFactory::Instance().create("Workspace2D", numberOfHistograms, numberOfChannels + 1, numberOfChannels);
-  m_localWorkspace->getAxis(0)->unit() = UnitFactory::Instance().create("Wavelength");
+void LoadILLSANS::createEmptyWorkspace(const size_t numberOfHistograms, const size_t numberOfChannels,
+                                       const MultichannelType type) {
+  m_localWorkspace = WorkspaceFactory::Instance().create(
+      "Workspace2D", numberOfHistograms, numberOfChannels + ((type == MultichannelType::TOF && !m_isD16Omega) ? 1 : 0),
```

Comment:
The decision whether to add one extra channel is a little busy as an parameter of creating a workspace. Could you extract it to a variable?